x/oracle/types/ballot.go

```diff
@@ -132,6 +133,26 @@ func (pb ExchangeRateBallot) Swap(i, j int) {
 	pb[i], pb[j] = pb[j], pb[i]
 }
 
+// BallotDenom is a convenience wrapper for setting rates deterministically.
+type BallotDenom struct {
+	Ballot ExchangeRateBallot
+	Denom  string
+}
+
+// BallotMapToSlice returns an array of sorted exchange rate ballots.
+func BallotMapToSlice(votes map[string]ExchangeRateBallot) (b []BallotDenom) {
+	for denom, ballot := range votes {
```

Comment:
we can make this slightly more efficient by:

```suggestion
func BallotMapToSlice(votes map[string]ExchangeRateBallot) ([]BallotDenom) {
	b := make([]BallotDenom, len(votes))

	for denom, ballot := range votes {
```